### DISCUSSION TITLE
SEI on cracks on SPM-Model

### DISCUSSION THREAD

**Discussion by fsenoner** [pybamm-team/PyBaMM#3297] [General]
Dear PyBaMM team,

I conducted a simulation using an SPM model. The submodel "SEI" was utilized with the following parameters: "ec reaction limited," "SEI on cracks" set to "true," and "particle mechanics" defined as ("swelling and cracking," "swelling only"). Several simulations were executed, wherein the cracking rate was varied. The results of the measured capacity and the loss of capacity due to SEI on cracks do not align, as evident in the plot.

After 1750 cycles, the "measured capacity" remains at 15Ah for all values of the cracking rate. However, in the plot "loss of capacity due to SEI on cracks," all values exceed 2.5Ah. Hence, the measured capacity should be lower. Could it be that this capacity loss is not being accounted for the calculation of the measured capacity?

![image](https://github.com/pybamm-team/PyBaMM/assets/130640867/a87ccbe0-c620-480f-8aee-23fb1effabc8)

Thank you very much for your time!

**Comment by brosaplanella** [comment]
@DrSOKane any thoughts on this?

**Comment by DrSOKane** [comment]
After much debugging, I tried adding `{"surface form": "algebraic"}` to the options and it works just fine. 

One of the advantages DFN has over the various SPM models is that current conservation is one of the fundamental DFN equations and is therefore guaranteed. Whereas for SPM models it has to be imposed with the surface form option. If you do not use the surface form, current is not conserved if you have side reactions. If you assume the side reactions are orders of magnitude slower than the main reactions, you can get away with not conserving current, as demonstrated by Ferran in [this paper](https://www.sciencedirect.com/science/article/pii/S0307904X22005959?via%3Dihub). Why it's such an issue for SEI on cracks I'm not sure.
